workspace/notebook/py_sb_harmonised_nsip_invoice.json

```diff
@@ -0,0 +1,644 @@
+						}
+					}
+				},
+				"source": [
+					"#### Import required libraries"
+				]
+			},
+			{
+				"cell_type": "code",
+				"source": [
+					"from pyspark.sql import functions as F\n",
+					"from pyspark.sql.window import Window\n",
+					"from datetime import datetime\n",
+					"from delta.tables import DeltaTable\n",
+					"from notebookutils import mssparkutils\n",
+					"from pyspark.sql.functions import explode, col"
+				],
+				"execution_count": null
+			},
+			{
+				"cell_type": "markdown",
+				"metadata": {
+					"nteract": {
+						"transient": {
+							"deleting": false
+						}
+					}
+				},
+				"source": [
+					"##### Configurable Variables for Service Bus and Spark Table Processing"
+				]
+			},
+			{
+				"cell_type": "code",
+				"source": [
+					"# Configurable variables\n",
+					"service_bus_table = \"odw_harmonised_db.sb_nsip_project\"\n",
+					"db_name=\"odw_harmonised_db\"\n",
+					"table_name=\"nsip_invoice\"\n",
+					"spark_table_final = f\"{db_name}.{table_name}\"\n",
+					"incremental_key = \"NSIPInvoiceID\"\n",
+					"start_exec_time = datetime.now()\n",
+					"insert_count = 0\n",
+					"update_count = 0\n",
+					"delete_count = 0\n",
+					"error_message=''"
+				],
+				"execution_count": null
+			},
+			{
+				"cell_type": "markdown",
+				"metadata": {
+					"nteract": {
+						"transient": {
+							"deleting": false
+						}
+					}
+				},
+				"source": [
+					"##### Read and Filter Harmonised Metadata for Delta Table Path"
+				]
+			},
+			{
+				"cell_type": "code",
+				"source": [
+					"\n",
+					"if not error_message:  # Only execute if there was no previous error\n",
+					"    try:\n",
+					"        storage_account: str = mssparkutils.notebook.run('/utils/py_utils_get_storage_account')\n",
+					"        metadata_path: str = f\"abfss://odw-config@{storage_account}/existing-tables-metadata.json\"\n",
+					"\n",
+					"        # Read JSON metadata\n",
+					"        df = spark.read.json(metadata_path, multiLine=True)\n",
+					"\n",
+					"        # Explode harmonised metadata\n",
+					"        harmonised_df = df.select(explode(col(\"harmonised_metadata\")).alias(\"metadata\"))\n",
+					"\n",
+					"        # Filter based on db_name and table_name\n",
+					"        filtered_df = harmonised_df.filter(\n",
+					"            (col(\"metadata.database_name\") == db_name) &\n",
+					"            (col(\"metadata.table_name\") == table_name)\n",
+					"        ).select(\n",
+					"            col(\"metadata.table_location\").alias(\"table_location\"),\n",
+					"            col(\"metadata.table_type\").alias(\"table_type\"),\n",
+					"            col(\"metadata.table_format\").alias(\"table_format\")\n",
+					"        )\n",
+					"\n",
+					"        # Show filtered result for debugging\n",
+					"        filtered_df.show(truncate=False)\n",
+					"\n",
+					"        # Assign table_location to delta_table_path\n",
+					"        rows = filtered_df.collect()\n",
```

Comment:
Could you read the `existing-tables-metadata.json` using Python's `json` module instead? I think that will be faster (although i've not done a performance test in synapse for this)